app/src/main/java/net/gsantner/markor/format/asciidoc/AsciidocTextConverter.java

```diff
@@ -0,0 +1,59 @@
+/*#######################################################
+ *
+ *   Maintained by Gregor Santner, 2018-
+ *   https://gsantner.net/
+ *
+ *   License of this file: Apache 2.0 (Commercial upon request)
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+#########################################################*/
+package net.gsantner.markor.format.asciidoc;
+
+import android.content.Context;
+
+import net.gsantner.markor.format.TextConverterBase;
+import net.gsantner.opoc.util.GsFileUtils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@SuppressWarnings("WeakerAccess")
+public class AsciidocTextConverter extends TextConverterBase {
+    //########################
+    //## Extensions
+    //########################
+
+    private static final List<String> EXT_ASCIIDOC = Arrays.asList(".adoc", ".asciidoc", ".asc");
+    private static final List<String> EXT = new ArrayList<>();
```

Comment:
Is AsciiDoc very similar to any other markup format? Like Zim and WikiText are very similar, so they both use the same format.

If so, please add it there to the mapping. You can also remove the static Ext.addAll, and just have EXT = new (adoc, ..).   
In example of PlainText converter it's used to run different conversion based on file extension - but this is not needed here.

Reply:
The main reason I started implementing it is: it doesn't fit with the other formats. All kinds of compromises would be necessary. Logic is different. And I've been trying to use Markor for several years now without this specific AsciiDoc support. And I was asked to learn Java, to contribute myself, and to maintain the AsciiDoc format. So I started to do that.

Reply:
You misunderstood. I asked if there is ANY other format you know is similar syntax wise to AsciiDoc. It doesn't relate to what Markor supports or not.

Reply:
I moved from Markdown to AsciiDoc. They are very similar:
https://docs.asciidoctor.org/asciidoc/latest/asciidoc-vs-markdown/

The main reason why I switched:
- **includes**: https://docs.asciidoctor.org/asciidoc/latest/directives/include/  
  I love them! They give me the flexibility I've always lacked in Markdown: I can separate content and usage, like in a content management system. For example, for database documentation, I can automatically create the content and use it separately. About my database documentation generator and how I use includes:
https://datahandwerk.gitlab.io/dhw/user-guide/documentation-generator.html#_how_does_it_work  
- There are not dozens of dialects, but exactly one standard
- build in TOC

Reply:
Again, sorry to say - you don't get it what and why I ask. You define a list of extensions, like .adoc, .asciidoc. My questoin is if there are other formats to asciidoc, which might be similar. So you would add .docbook for example to that asciidoc mapping as well. (Docbook is not similar, but just saying that as example).

Reply:
IMHO - other formats are different and should be implemented differently. I don't know other formats, which could match. That's why I included only AsciiDoc.

Reply:
>  I don't know other formats, which could match.

OK that was it all about 😄 .